nft_ingester/src/backfiller.rs

```diff
@@ -0,0 +1,702 @@
+                                        println!("Successfully backfilled tree: {tree_str}");
+
+                                        // Only delete extra tree rows if fetching and plugging gaps worked.
+                                        if let Err(err) = self
+                                            .delete_extra_rows_and_mark_as_backfilled(
+                                                &tree, max_seq,
+                                            )
+                                            .await
+                                        {
+                                            println!("Error deleting rows and marking as backfilled: {err}");
+                                            self.sleep_and_increase_delay().await;
+                                        } else {
+                                            // Debug.
+                                            println!("Successfully deleted rows up to {max_seq}");
+                                            self.reset_delay()
+                                        }
+                                    } else {
+                                        // Debug.
+                                        println!("Unexpected error, tree was in list, but no rows found for {tree_str}");
+                                        self.sleep_and_increase_delay().await;
+                                    }
+                                }
+                                Err(err) => {
+                                    println!("Failed to fetch and plug gaps for {tree_str}");
+                                    println!("{err}");
+                                    self.sleep_and_increase_delay().await;
+                                }
+                            }
+                        }
+                    }
+                }
+                Err(err) => {
+                    // Print error but keep trying.
+                    println!("Could not get trees to backfill from db: {err}");
+                    self.sleep_and_increase_delay().await;
+                }
+            }
+        }
+    }
+
+    async fn sleep_and_increase_delay(&mut self) {
+        sleep(Duration::from_millis(self.failure_delay)).await;
+
+        // Increase failure delay up to `MAX_FAILURE_DELAY_MS`.
+        self.failure_delay = self.failure_delay.saturating_mul(2);
+        self.failure_delay = if self.failure_delay > MAX_FAILURE_DELAY_MS {
+            MAX_FAILURE_DELAY_MS
+        } else {
+            self.failure_delay
+        };
+    }
+
+    fn reset_delay(&mut self) {
+        self.failure_delay = INITIAL_FAILURE_DELAY;
+    }
+
+    async fn get_trees_to_backfill(&self) -> Result<Vec<BackfillTree>, DbErr> {
+        // Get trees with the `force_chk` flag set.
+        let force_chk_trees = UniqueTree::find_by_statement(Statement::from_sql_and_values(
+            DbBackend::Postgres,
+            "SELECT DISTINCT backfill_items.tree FROM backfill_items\n\
```

Comment:
I used to have `r#` doc strings here, but switched to standard string so that I could do multi line with indenting.  When I used `r#` the indents for the multi line were included in the string.